File: OlypicGamesC++/Team.cpp
#include "Team.h"

vector<Athlete*>& Team::getVectorOfAthletes() {
	return athletes;
}

ostream& operator<<(ostream& os, const Team* team) {
	int i = 0;
	os << "[ ";
	for (Athlete* ath : team->athletes) {
		if (i != 0)os << ", ";
		os  << ath->getName();
		i++;
	}
	return os << " ]" << endl;
}

File: OlympicGamesJava/logic/Team.java
package logic;

import java.util.Vector;

public class Team extends Competitor {
	private Vector<Athlete> athletes = new Vector<Athlete>();
	
	public Team(Event e, String medal) {
		super(e, medal);
	}
	
	public void addAthlete(Athlete athlete) {
		athletes.add(athlete);
	}
	
	public int getSize() { return athletes.size(); }
	
	@Override
	public Vector<Athlete> getVectorOfAthletes() { return athletes; }
}
